server/app/services/application/ApplicationEventDetails.java

```diff
@@ -0,0 +1,101 @@
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import java.util.Optional;
+
+/**
+ * The details of a single Application event based on its type.
+ *
+ * <p>This class is intended to represent the specific relevant details of some event related to an
+ * {@code Application}. The event is identified by {code EventType} and its corresponding data; as
+ * such only one of the Optional detail fields should be present. This class will be json serialized
+ * into a database column.
+ */
+@AutoValue
+@JsonDeserialize(builder = AutoValue_ApplicationEventDetails.Builder.class)
+public abstract class ApplicationEventDetails {
+  /** The event affecting the application. */
+  @JsonProperty("event_type")
+  public abstract Type eventType();
+
+  // Only one of the following Event fields should be set.
+  // The JsonInclude suppresses empty Optionals from being serialized.
+  /** Type STATUS_EVENT representing a change in the Status Tracking status value. */
+  @JsonInclude(Include.NON_EMPTY)
+  @JsonProperty("status_event")
+  public abstract Optional<StatusEvent> statusEvent();
+
+  @JsonInclude(Include.NON_EMPTY)
+  @JsonProperty("note_event")
+  /** Type NOTE_EVENT representing a note associated with the Application. */
+  public abstract Optional<NoteEvent> noteEvent();
+
+  public static Builder builder() {
+    return new AutoValue_ApplicationEventDetails.Builder();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    @JsonProperty("event_type")
+    public abstract Builder setEventType(Type type);
+
+    @JsonProperty("status_event")
+    public abstract Builder setStatusEvent(StatusEvent event);
+
+    @JsonProperty("note_event")
+    public abstract Builder setNoteEvent(NoteEvent event);
+
+    public abstract ApplicationEventDetails build();
+  }
+
+  // The type of Event being recorded.
+  public enum Type {
+    STATUS_CHANGE,
+    NOTE_CHANGE;
+  }
+
+  @AutoValue
+  @JsonDeserialize(builder = AutoValue_ApplicationEventDetails_StatusEvent.Builder.class)
+  public abstract static class StatusEvent {
+    /**
+     * The text of the StatusDefinitions.Status applied to the application in the default locale.
+     */
+    @JsonProperty("status_text")
+    public abstract String statusText();
+    /** If the status has email content and if it was sent as part of setting the status. */
+    @JsonProperty("email_sent")
+    public abstract Boolean emailSent();
+
+    public static Builder builder() {
+      return new AutoValue_ApplicationEventDetails_StatusEvent.Builder();
+    }
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+      @JsonProperty("status_text")
+      public abstract Builder setStatusText(String statusText);
+
+      @JsonProperty("email_sent")
+      public abstract Builder setEmailSent(Boolean emailSent);
+
+      public abstract StatusEvent build();
+    }
+  }
+
+  @AutoValue
+  public abstract static class NoteEvent {
```

Comment:
As I understand right now (for V1), we're only going to display a single note in the application UI and not display the list of notes. Just wanted to check in how you see that being modeled here.

Reply:
This table is just a log, we can apply business logic on top of it.


As is, the most recent value is the relevant one, and the previous are history which we might surface later.

if/when we want to have multiple notes, we'd likely need to add a note id here and track that through them, we can write a data migrator to go update all the existing notes to have the same ID to bootstrap.

Reply:
Ack!